### PR TITLE
fix: remove style modifying all host components with hidden attribute

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7325

## Summary

- Removes style modifying all host components with hidden attribute. This is modifying components outside of calcite and applying host styles outside of the host. Host styles should have no effect outside of a shadow DOM. https://developer.mozilla.org/en-US/docs/Web/CSS/:host
- Adds includes mixin for common component internal styles such as hidden attribute styling
- Includes mixin in every component file.

### CODE REVIEW

File: packages/calcite-components/src/assets/styles/includes.scss [hunk 1]
```diff
@@ -75,6 +71,16 @@
   }
 }
 
+@mixin calciteComponentCommon() {
```
Comment (jcfranco) [inline packages/calcite-components/src/assets/styles/includes.scss:74]: We don't prefix other mixins w/ `calciteComponent`. Maybe `base` or `common`?

**Review by jcfranco — APPROVED** [review]
👁️🚫👻

File: packages/calcite-components/src/assets/styles/includes.scss (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/assets/styles/includes.scss:74]: I also noticed some mixins use camel case vs dash separated? do we want to pick one?

- calciteHydratedHidden
- xButton
- hidden-form-input
- word-break

Comment (driskull) [reply packages/calcite-components/src/assets/styles/includes.scss]: Renamed to `baseComponent`

**Pull request merged**